src/ui_basic_flows_nbgl.c

```diff
@@ -0,0 +1,251 @@
+#ifdef HAVE_NBGL
+#include <stdbool.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdint.h> 
+#include <os_io_seproxyhal.h>
+#include <ux.h>
+#include "ui_helpers.h"
+#include "glyphs.h"
+#include "nbgl_page.h"
+#include "nbgl_use_case.h"
+#include "menu.h"
+
+// Globals
+static ui_callback_fn_t * confirm_clbk;
+static ui_callback_fn_t * reject_clbk;
+static char* rejection_text;
+static nbgl_layoutTagValueList_t pairList;
+static nbgl_layoutTagValue_t fields[MAX_FIELDS_PER_TRANSACTION];
+static nbgl_pageInfoLongPress_t infoLongPress;
+static bool reviewStarted;
+static bool isDisplayAddress;
+static bool isDisplayPath;
+
+// Static functions declarations
+static void rejectUseCaseChoice(void);
+static void displayTransaction(void);
+static void rejectAndDisplayIdleMenu(void);
+static void displayAddress(void);
+
+// Static functions definitions
+static void warningChoice(bool accept) {
+    if (accept) {
+        confirm_clbk();
+    }
+    else {
+        rejectAndDisplayIdleMenu();
+    }
+}
+
+static void contractWarningChoice(bool accept) {
+    if (accept) {
+        displayTransaction();
+    }
+    else {
+        rejectAndDisplayIdleMenu();
+    }
+}
+
+
+static void reviewChoice(bool confirm) {
+  if (confirm) {
+    confirm_clbk();
+    nbgl_useCaseStatus("TRANSACTION\nCONFIRMED",true,ui_idle);
+  }
+  else {
+    rejectUseCaseChoice();
+  }
+}
+
+static void rejectChoice(bool confirm) {
+  if (confirm){
+    rejectAndDisplayIdleMenu();
+  }
+  else{
+    if(reviewStarted)
+    {
+      displayTransaction();
+    }
+    else
+    {
+      ui_reviewStart(confirm_clbk,reject_clbk);
+    }
+  }
```

Comment:
Can't the modal do this ?

Reply:
Yes it can. I think I used this "trick" because `nbgl_useCaseChoice` wasn't yet available. I'll update the code.

Reply:
Done.